server/src/main/java/org/opensearch/search/query/QuerySearchResult.java

```diff
@@ -364,6 +367,11 @@ public void readFromWithId(ShardSearchContextId id, StreamInput in) throws IOExc
         nodeQueueSize = in.readInt();
         setShardSearchRequest(in.readOptionalWriteable(ShardSearchRequest::new));
         setRescoreDocIds(new RescoreDocIds(in));
+        if (in.getVersion().onOrAfter(minimumTookTimeVersion)) {
```

Comment:
No need to define minimumTookTimeVersion at a class level as any future changes/variables will have their own version. You should directly use Version.V_3_0_0 here and below.